alchemist-graphql/src/jvmMain/kotlin/it/unibo/alchemist/boundary/launchers/GraphQLServerLauncher.kt

```diff
@@ -20,21 +25,37 @@ import kotlinx.coroutines.runBlocking
 /**
  * A launcher that starts a GraphQL server for exposing the simulation.
  */
-class GraphQLServerLauncher : SimulationLauncher() {
+class GraphQLServerLauncher @JvmOverloads constructor(
+    private val port: Int = 8081,
+    private val host: String = "127.0.0.1",
+) : SimulationLauncher() {
     override fun launch(loader: Loader) {
         val simulation: Simulation<Any, Nothing> = prepareSimulation(loader, emptyMap<String, Any>())
         startServer(simulation)
     }
 
-    private fun startServer(
-        simulation: Simulation<Any, Nothing>,
+    private fun <T, P : Position<P>> startServer(
+        simulation: Simulation<T, P>,
         serverDispatcher: CoroutineDispatcher = Dispatchers.IO,
     ) {
         return runBlocking {
             launch(serverDispatcher) {
-                EngineMain.main(emptyArray())
+                val server = makeServer(simulation)
+                server.start(wait = true)
             }
             simulation.run()
         }
     }
+
+    private fun<T, P : Position<out P>> makeServer(simulation: Simulation<T, P>) =
+        embeddedServer(
+            Netty,
+            port = port,
+            host = host,
+            module = {
+                attributes.put(SimulationAttributeKey, simulation)
+                graphQLModule()
+                graphQLRoutingModule()
+            },
+        )
 }
```

Comment:
Can you remember me why you're using an `embeddedServer` instead on of the ktor Engine? As far as I remember using the other method is considered the "best practice", but maybe I'm just forgetting something.

Reply:
Is it to pass the Simulation? Can you please check if it is possible to do so using the other method too?

Reply:
Ok, this can be considered resolved for me.

Reply:
@DanySK can i mark this as resolved?

Reply:
I don't know why but I could'nt. Maybe my UI was bugged, or (even more) probably i'm just blind.